rmgpy/reaction.py

```diff
@@ -1171,16 +1171,11 @@ def generate_pairs(self):
             def get_sorting_key(spc):
                 # List of elements to sort by, order is intentional
                 numbers = [6, 8, 7, 14, 16, 15, 17, 53, 9, 35]  # C, O, N, Si, S, P, Cl, I, F, Br
-                num_ele = []
-                for n in numbers:
-                    ele = 0
-                    for atom in spc.molecule[0].atoms:
-                        if isinstance(atom, Atom):
-                            if atom.element.number == n:
-                                ele += 1
-                    num_ele.append(ele)
-                # tuple(sum([1 for atom in spc.molecule[0].atoms if isinstance(atom,Atom) and atom.element.number == n]) for n in numbers)
-                return tuple(num_ele)
+                ele_count = dict([(n,0) for n in numbers])
+                for atom in spc.molecule[0].atoms:
+                    if isinstance(atom, Atom) and atom.element.number in numbers:
+                        ele_count[atom.element.number] += 1
+                return tuple(ele_count.values())
```

Comment:
Thanks for changing this! I think this is good, but just to make sure that the tuple preserves the order, can you see if something like the following works?

```
return (ele_count[n] for n in numbers)
```

Reply:
Done!